### ISSUE TITLE
Remove support for python 2.7

### ISSUE THREAD

**Issue by compilenix** [proxmoxer/proxmoxer#13]
Python 2.7 is officially [end-of-life since Jan 1, 2020](https://www.python.org/doc/sunset-python-2/).

So, i would argue that we should remove support for python 2.7 in the near feature.

**Comment by jhollowe** [comment]
While I agree Python 2 should disappear, that takes effort from the entire community. While I would encourage all development in the future to prioritize 3.x, removing existing Python 2 support doesn't make sense. There is still likely 2.7 code that is using this library which we should not force to update until there is a reason to do so.

If there is a feature we would like to implement in the future that requires 3.x I would push to include it and end support for 2.x. However, code that does not *require* 3 should continue using existing cross-version code.

If you have a good reason to end 2.7 support, I would love to hear it and consider it, but as I see it, we don't *need* to do so at the current time.

TL;DR I would like to keep 2.x until there is a code-base reason to remove support.

**Comment by jhollowe** [comment]
I'm going to close this, but anyone one in the future feel free to comment
